File: csharp/transpose/Transpose.cs
public static class Transpose
{
    public static string String(string input)
    {
        string[] rows = input.Split("\n");
        string result = "";
        int maxRowLength = 0;
        foreach (string row in rows)
        {
            if (row.Length >= maxRowLength)
            {
                maxRowLength = row.Length;
            }
        }
        for (int i = 0; i < maxRowLength; i++)
        {
            for (int j = 0; j < rows.Length; j++)
            {
                string row = rows[j];
                if (i < row.Length)
                {
                    result += row[i];
                }
                else
                {
                    if (rows[j..].Any(row => i < row.Length))
                    {
                        result += " ";
                    }
                }
            }
            if (i < maxRowLength - 1)
            {
                result += "\n";
            }
        }
        return result;
    }
}

File: typescript/transpose/transpose.ts
export function transpose(input: string[]): string[] {
  if (input.length === 0) return [];
  const maxLength: number = Math.max(...input.map(row => row.length));
  return Array.from({ length: maxLength}, (_, colIndex: number) => {
    return input.reduceRight((acc: string, row: string) => {
      const char: string | undefined = row[colIndex];
      if (char !== undefined) {
        return char + acc;
      }
      return (acc ? ' ' : '') + acc;
    }, '');
  });
}
